wheatley/row_generation/go_and_stop_calling_generator.py

```diff
@@ -2,36 +2,29 @@
 
 # pylint: disable=protected-access
 
-import random
-
 from wheatley import calls
-from wheatley.aliases import Row
+from wheatley.aliases import Row, RowCall
 from wheatley.stroke import Stroke
-from wheatley.tower import RingingRoomTower
 
 from .row_generator import RowGenerator
 
 
 class GoAndStopCallingGenerator(RowGenerator):
     """ A decorator to add calling go and stop to an existing RowGenerator. """
 
-    def __init__(self, generator: RowGenerator, tower: RingingRoomTower) -> None:
+    def __init__(self, generator: RowGenerator) -> None:
         super().__init__(generator.stage)
 
-        self.tower = tower
         self.generator = generator
         self.called_go = False
 
-    def next_row(self, stroke: Stroke) -> Row:
-        if not self.called_go and stroke.is_back() and random.choices([True, False], [1, 3]):
-            self.tower.make_call(calls.GO)
```

Comment:
It is currently not possible for `RowGenerator`s to call 'Go' (this happened since we stopped having `RowGenerator`s generate the rounds at the start of ringing).  This also neatly removes the interdependence between the `RowGenerator`s and the implementation of the `Tower` class.

However, I will add the ability for row generators to specify that calls should be made **before** the rows start, along with which stroke the ringing starts (so that backstroke starts are correctly handled).

There's currently a bug in CompLib (GACJ/complib#194) that prevents bobs happening before the ringing starts from appearing anyway, so there isn't really a way to test this for `Bob`s or `Single`s.